### PR TITLE
[Next.js][Multi-site] Dynamic site resolver

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description / Motivation
* Introduced `SiteResolver`
* Returns `siteNamed` based on the provided `hostName` and `siteInfo` list
* Returns fallback default/custom site name when appropriate `siteInfo` is not found

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by art-alexeyenko — CHANGES_REQUESTED** [review]

File: packages/sitecore-jss/src/site/site-resolver.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// This resolver is used to determine site name based on the provided hostname
+export class SiteResolver {
+  /**
+   * Resolve siteName by hostName
+   * @param {string} hostName
+   * @param {SiteData[]} sitesInfo
+   * @param {string} [fallbackSiteName] siteName to be returned in case siteName is not found
+   * @returns {string} siteName
+   */
+  static resolve = (hostName: string, sitesInfo: SiteData[], fallbackSiteName?: string): string => {
```
Comment (ambrauer) [inline packages/sitecore-jss/src/site/site-resolver.ts:17]: I was expecting the site resolver would perform the fetching of the `SiteData[]` as well (as opposed to a param) which should be possible once the `GraphQLSiteInfoService` is merged from [this PR](https://github.com/Sitecore/jss/pull/1227). However, I can see the benefit to keeping these separate.

What do you thing @art-alexeyenko? Probably a larger discussion we should have... Or something we can always adjust as POC moves along.

File: packages/sitecore-jss/src/site/site-resolver.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    const siteInfo = sitesInfo.find((info) => info.hostName === hostName);
```
[the remark this answers is not in the document]
Comment (illiakovalenko) [reply packages/sitecore-jss/src/site/site-resolver.ts:13]: how it should be resolved in this case? which sitename should be returned, or fallback should be provided?

File: packages/sitecore-jss/src/site/site-resolver.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/sitecore-jss/src/site/site-resolver.ts]: I chose this approach, cause I thought that site-resolver can be used in multiple places

Comment (art-alexeyenko) [reply packages/sitecore-jss/src/site/site-resolver.ts]: I don't see much use in anything from SiteData/SiteInfo other than the site name, after it was resolved. Every other piece of data (hostname, language etc) that we may have there would already be known and serve as input into the resolve method.

**Review by ambrauer — APPROVED** [review]
Approving for now. As discussed, we can continue in follow-up items.

**Pull request merged**